### PR TITLE
Feature/64777

### PR DESCRIPTION
**Pull request by anqaka**
internal DEV-64777

### CODE REVIEW

File: Magento_Catalog/layout/catalog_product_view.xml [hunk 1]
```diff
@@ -289,6 +289,49 @@
...
         </referenceBlock>
 
+        <referenceBlock name="catalog.product.related">
+            <block
+                class="Magento\Framework\View\Element\RendererList"
+                name="related.product.type.details.renderers"
+                as="details.renderers"
+            >
+                <block class="Magento\Framework\View\Element\Template" as="default"/>
+                <block class="Magento\Swatches\Block\Product\Renderer\Listing\Configurable" name="related.product.type.details.renderers.configurable" as="configurable" template="Magento_Swatches::product/listing/renderer.phtml">
```
Comment (pawel-snowdog) [inline Magento_Catalog/layout/catalog_product_view.xml:299]: formatting

File: Magento_Catalog/templates/product/list/items.phtml [hunk 2]
```diff
@@ -319,93 +328,64 @@ $sliderResponsiveConfig = '[
                                         <?php if ($showCart): ?>
                                             <?php if ($_item->isSaleable()): ?>
                                                 <div class="product-grid-item__primary-action">
-                                                    <?php if ($_item->getTypeInstance()->hasRequiredOptions($_item)): ?>
+                                                    <!-- Adding to cart with ajax when product has required option and is available -->
+                                                    <?php
+
```
Comment (pawel-snowdog) [inline Magento_Catalog/templates/product/list/items.phtml:333]: emty space

File: Magento_Catalog/layout/catalog_product_view.xml [hunk 3]
```diff
@@ -289,6 +289,49 @@
...
+                    <arguments>
+                        <argument name="configurable_view_model" xsi:type="object">Magento\Swatches\ViewModel\Product\Renderer\Configurable</argument>
```
Comment (pawel-snowdog) [inline Magento_Catalog/layout/catalog_product_view.xml:301]: formatting

File: Magento_Catalog/layout/catalog_product_view.xml (hunk 1 above)
Comment (anqaka) [reply Magento_Catalog/layout/catalog_product_view.xml:299]: fixed

File: Magento_Catalog/templates/product/list/items.phtml (hunk 2 above)
Comment (anqaka) [reply Magento_Catalog/templates/product/list/items.phtml:333]: fixed

File: Magento_Catalog/layout/catalog_product_view.xml (hunk 3 above)
Comment (anqaka) [reply Magento_Catalog/layout/catalog_product_view.xml:301]: fixed

**Pull request merged**